### ISSUE TITLE
prepared_for_logs and stored procedures

### ISSUE THREAD

**Issue by ghost** [sufleR/sql_query#3]
Hi, and many thanks for `sql_query`.  I'm using it to load stored procedures into my PostGIS DB for a rails app.  However, I find that I get postgres syntax errors on my call

```
SqlQuery.new(:procedure_cell_truncate, opts).execute
```

specifically

```
ActiveRecord::StatementInvalid: PG::SyntaxError: ERROR:  syntax error at end of input
LINE 1: ...ygon FROM geom ) AS q WHERE cells.id = q.id; END; $body$ LAN...
```

(The full `.sql.erb` file is [here](https://gist.github.com/jjgreen/3a618920ee2bd90c8d84)).

I found that I could pipe the output of     

```
SqlQuery.new(:procedure_cell_truncate, opts).sql
```

directly to `psql` and there was no syntax error.  So on a hunch, I monkey-patched the `prepared_for_logs` method with 

```
class SqlQuery
  def prepared_for_logs
    sql
  end
end
```

and then the syntax error went away.

I'm really not sure why the `gsub` in `prepared_for_logs` is borking this query, perhaps this is obvious to someone else?

**Comment by ghost** [comment]
Ah, I think I see the problem: the `--` comment lines.  I find that modifying my monkey-patch to

```
class SqlQuery
  def prepared_for_logs
    sql
      .gsub(/--.*\n/, ' ')
      .gsub(/(\n|\s)+/,' ')
  end
end
```

seems to fix the problem; possibly there is a better way of doing the same thing :smile:

**Comment by sufleR** [comment]
If you want just comment your code it will be better to use ruby comments

```
<% # comment here %>
```

I'm not sure what expected behaviour should look like.
Do you want to have comments in sql? 
If yes execute shouldn't use `prepared_for_logs` at all.

I think best solution will be to add special option to keep sql query unchanged during execution.

Example

```
SqlQuery.new(:procedure_cell_truncate, opts.merge(prepare_for_logs: false)).execute
```

I implemented solution above already.
Check latest commit.
If it will satisfy your needs I will release new version with it.

**Comment by ghost** [comment]
Ah cool, thanks!

One suggestion, perhaps the option could be passed to `.execute` rather than the constructor?  It seems to violate the "principle of least surprise" that `.prepare_for_logs` do anything but prepare the query for logs.  Perhaps

```
def execute(opts = {})
  unless opts.has_key? :prepare_for_logs then
    opts[:prepare_for_logs] = true
  end
  sql_to_execute =
    if opts[:prepare_for_logs] then
       prepared_for_logs
    else
       sql
    end
  connection.execute(sql_to_execute).entries
end
```

**Comment by sufleR** [comment]
much better idea.

Changed to pass prepare argument to execute

```
SqlQuery.new(:procedure_cell_truncate, opts).execute(false)
```

will use unchanged sql instead of prepared one.

**Comment by ghost** [comment]
Excellent, thank you for the modification (and the gem itself, elegant).

**Comment by sufleR** [comment]
released in 0.4.0
